pepper-apis/dss-core/src/test/java/org/broadinstitute/ddp/route/IntegrationTestSuite.java

```diff
@@ -396,11 +362,47 @@ private static boolean isTestServerRunning() {
 
     }
 
+    /**
+     * Called by circleci when creating a dss backend
+     * to run tests against.  Starting a test instance
+     * from scratch can take 30s or more, during which
+     * time most routes will not work properly.  To check
+     * for readiness, ping port 5999 with a utility
+     * like netcat.  When you can connect to port 5999,
+     * the instance is ready for traffic.  Note that
+     * this administrative port is only available for one
+     * connection.  After that, the port closes.
+     */
     public static void main(String[] args) {
         initializeDatabase();
         startupTestServer(true);
         insertTestData();
         TransactionWrapper.useTxn(TransactionWrapper.DB.APIS, LanguageStore::init);
+        // now start a separate service that circleCI can ping once
+        // all the test data has been loaded and client tests can run
+        new SingleUseServerSocket(5999);
+    }
+
+    /**
+     * Server socket that allows one connection
+     * and then closes
+     */
+    public static class SingleUseServerSocket {
+
+        public SingleUseServerSocket(int port) {
+            ServerSocket serverSocket = null;
```

Comment:
This appears a bit clunky at first--why not just use another spark port?  The answer is that making multiple spark services on different ports in the same VM requires more extensive changes, and also requires all future new routes to reference a particular spark service instead of just `get(...), post(...)` etc.  It would also require changes to our circleCI env to check curl return codes and result bodies.  Since circleCI is already doing simple TCP connections via `nc`, I decided to stick with that pattern and make a simple port available here for circleCI.